cqf-fhir-cr/src/main/java/org/opencds/cqf/fhir/cr/visitor/PackageVisitor.java

```diff
@@ -184,15 +195,34 @@ protected void handleValueSets(IBaseBundle packagedBundle, Optional<IBaseResourc
                 .filter(r -> r.fhirType().equals("ValueSet"))
                 .map(v -> (IValueSetAdapter) createAdapterForResource(v))
                 .collect(Collectors.toList());
-
-        valueSets.stream()
-                .forEach(valueSet -> expandHelper.expandValueSet(
+        var expansionCache = this.getExpansionCache();
+        var expansionParamsHash = expansionCache.map(
+                e -> e.getExpansionParametersHash(rootSpecificationLibrary).orElse(null));
+        if (expansionCache.isPresent()) {
+            valueSets.forEach(v -> {
+                var cachedExpansion = expansionCache
+                        .get()
+                        .getExpansionForCanonical(v.getCanonical(), expansionParamsHash.orElse(null));
+                if (cachedExpansion != null) {
+                    v.setExpansion(cachedExpansion.getExpansion());
+                    expandedList.add(v.getUrl());
+                }
+            });
+        }
+        valueSets.stream().forEach(valueSet -> {
```

Comment:
We are iterating over each ValueSet twice here if we have an expansion cache.  This should be refactored to iterate only a single time.

Reply:
I see.  Yeah, that seems reasonable.